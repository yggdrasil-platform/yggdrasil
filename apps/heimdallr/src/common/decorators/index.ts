export { default as CurrentUser } from './CurrentUser';
