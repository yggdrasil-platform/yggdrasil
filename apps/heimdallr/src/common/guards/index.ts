export { default as GqlAuthGuard } from './GqlAuthGuard';
export { default as LocalAuthGuard } from './LocalAuthGuard';
