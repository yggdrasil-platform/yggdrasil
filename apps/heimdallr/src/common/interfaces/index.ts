export { default as IEnvironmentVariables } from './IEnvironmentVariables';
export { default as IRequest } from './IRequest';
