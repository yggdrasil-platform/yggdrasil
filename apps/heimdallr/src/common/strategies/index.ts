export { default as JwtStrategy } from './JwtStrategy';
export { default as LocalStrategy } from './LocalStrategy';
