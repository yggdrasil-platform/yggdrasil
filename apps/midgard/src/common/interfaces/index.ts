export type { default as IEnvironmentVariables } from './IEnvironmentVariables';
