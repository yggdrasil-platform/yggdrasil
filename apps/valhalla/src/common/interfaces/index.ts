export { default as IUserServiceEnvironmentVariables } from './IUserServiceEnvironmentVariables';
