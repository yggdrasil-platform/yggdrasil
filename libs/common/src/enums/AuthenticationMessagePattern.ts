enum AuthenticationMessagePattern {
  Authenticate = 'Authenticate',
}

export default AuthenticationMessagePattern;
