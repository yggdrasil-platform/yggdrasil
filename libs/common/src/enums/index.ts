export { default as DefaultMessagePatterns } from './DefaultMessagePatterns';
export { default as Providers } from './Providers';
export { default as Routes } from './Routes';
export { default as UserMessagePatterns } from './UserMessagePatterns';
