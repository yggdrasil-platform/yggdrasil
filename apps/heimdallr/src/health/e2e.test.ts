import { HttpStatus } from '@nestjs/common';
import { agent as request, Response, SuperAgentTest } from 'supertest';

// Enums
import { Routes } from '@libs/common/enums';

describe(__filename, () => {
  let agent: SuperAgentTest;

  beforeAll(() => {
    agent = request(
      `http://localhost:${process.env.EXTERNAL_APP_PORT || 3001}`,
    );
  });

  describe(`GET ${Routes.Health}`, () => {
    it('should connect to all the other services successfully', async () => {
      const response: Response = await agent
        .get(Routes.Health)
        .expect(HttpStatus.OK);

      expect(response.body).toMatchSnapshot();
    });
  });
});
