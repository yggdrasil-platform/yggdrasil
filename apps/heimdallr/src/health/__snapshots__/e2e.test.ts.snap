// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/home/kieran/code/repos/yggdrasil-platform/yggdrasil/apps/heimdallr/src/health/e2e.test.ts GET /health should connect to all the other services successfully 1`] = `
Array [
  Object {
    "environment": "test",
    "health": Object {
      "details": Object {
        "database": Object {
          "status": "up",
        },
      },
      "error": Object {},
      "info": Object {
        "database": Object {
          "status": "up",
        },
      },
      "status": "ok",
    },
    "name": "mimir",
    "version": "1.1.0",
  },
  Object {
    "environment": "test",
    "health": Object {
      "details": Object {
        "database": Object {
          "status": "up",
        },
      },
      "error": Object {},
      "info": Object {
        "database": Object {
          "status": "up",
        },
      },
      "status": "ok",
    },
    "name": "valhalla",
    "version": "1.1.0",
  },
]
`;
